dialogs.c

```diff
@@ -585,6 +587,14 @@ static void refresh_usb(void)
 
 	/* Fill things in */
 	connect_clear(dialogs.connect_usb);
+	gdk_threads_leave();
+
+}
+
+
+static void refresh_usb(void)
```

Comment:
I'm also a bit puzzled. The commit message mentions network backend but the code seems to "say" usb?

Reply:
The old iio-osc function responsible for scanning the context list was "refresh_usb". In fact, it handles all the IIO backends based on the available filter from GUI (network, local, usb, serial).
In order to reduce the number of changes to the iio-osc code (which I thought would bring more bugs and confusion to the table) I reused the old function name: "refresh_usb". 

If you think it's necessary, I can rename all these functions and all the calls to it to something related to context scanning.

The commit message mentions network because from my tests, scanning only blocks on network scanning. There is no issue on MacOS if we force iio-osc to only scan USB backends for example. What happens is that from the iio-osc main thread, libiio scanning gets up to these points: https://github.com/analogdevicesinc/libiio/blob/libiio-v0/dns_sd_bonjour.c#L184  https://github.com/analogdevicesinc/libiio/blob/libiio-v0/lock.c#L42 and then hangs. This seems to be due to these gdk thread guards: https://github.com/analogdevicesinc/iio-oscilloscope/blob/main/oscmain.c#L245-L261 .
Libiio context scanning works fine on MacOS outside of iio-osc (in iio_info for example).


The FMCOMMS2 advanced plugin calibration thread handles this in the same way I proposed for scanning: https://github.com/analogdevicesinc/iio-oscilloscope/blob/main/plugins/fmcomms2_adv.c#L830 .